tests/v2/validation/upgrade/kubernetes.go

```diff
@@ -1,151 +1,188 @@
-func validateKubernetesVersions(t *testing.T, client *rancher.Client, bc *bundledclusters.BundledCluster, versionToUpgrade *string, isUsingCurrentCluster bool) {
-	t.Helper()
+			clusterResp, err := clusters.GetClusterIDByName(client, upgradedCluster.Name)
+			require.NoError(u.T(), err)
 
-	cluster, err := bc.Get(client)
-	require.NoErrorf(t, err, "[%v]: Error occurred while validating kubernetes version", bc.Meta.Name)
+			upgradedRKE1Cluster, err := client.Management.Cluster.ByID(clusterResp)
+			require.NoError(u.T(), err)
 
-	if isUsingCurrentCluster {
-		cluster = bc
-	}
+			provisioning.VerifyRKE1Cluster(u.T(), client, testConfig, upgradedRKE1Cluster)
+		} else {
+			upgradedCluster, err := upgradeRKE2K3SCluster(u.T(), client, cluster, testConfig)
+			require.NoError(u.T(), err)
 
-	switch cluster.Meta.Provider {
-	case clusters.KubernetesProviderRKE:
-		if cluster.Meta.IsImported {
-			assert.Equalf(t, *versionToUpgrade, cluster.V3.RancherKubernetesEngineConfig.Version, "[%v]: %v", cluster.Meta.Name, logMessageKubernetesVersion)
-		} else if !cluster.Meta.IsImported {
-			assert.Equalf(t, *versionToUpgrade, cluster.V3.RancherKubernetesEngineConfig.Version, "[%v]: %v", cluster.Meta.Name, logMessageKubernetesVersion)
-		}
-	case clusters.KubernetesProviderRKE2:
-		if cluster.Meta.IsImported {
-			clusterSpec := &apiv1.ClusterSpec{}
-			err = v1.ConvertToK8sType(cluster.V1.Spec, clusterSpec)
-			require.NoError(t, err)
-
-			assert.Equalf(t, *versionToUpgrade, clusterSpec.KubernetesVersion, "[%v]: %v", cluster.Meta.Name, logMessageKubernetesVersion)
-		} else if !cluster.Meta.IsImported {
-			assert.Equalf(t, *versionToUpgrade, cluster.V3.Rke2Config.Version, "[%v]: %v", cluster.Meta.Name, logMessageKubernetesVersion)
-		}
-	case clusters.KubernetesProviderK3S:
-		if cluster.Meta.IsImported {
-			clusterSpec := &apiv1.ClusterSpec{}
-			err = v1.ConvertToK8sType(cluster.V1.Spec, clusterSpec)
-			require.NoError(t, err)
-
-			assert.Equalf(t, *versionToUpgrade, clusterSpec.KubernetesVersion, "[%v]: %v", cluster.Meta.Name, logMessageKubernetesVersion)
-		} else if !cluster.Meta.IsImported {
-			assert.Equalf(t, *versionToUpgrade, cluster.V3.K3sConfig.Version, "[%v]: %v", cluster.Meta.Name, logMessageKubernetesVersion)
+			provisioning.VerifyCluster(u.T(), client, testConfig, upgradedCluster)
 		}
-	case clusters.KubernetesProviderGKE:
-		assert.Equalf(t, *versionToUpgrade, *cluster.V3.GKEConfig.KubernetesVersion, "[%v]: %v", cluster.Meta.Name, logMessageKubernetesVersion)
-	case clusters.KubernetesProviderAKS:
-		assert.Equalf(t, *versionToUpgrade, *cluster.V3.AKSConfig.KubernetesVersion, "[%v]: %v", cluster.Meta.Name, logMessageKubernetesVersion)
-	case clusters.KubernetesProviderEKS:
-		assert.Equalf(t, *versionToUpgrade, *cluster.V3.EKSConfig.KubernetesVersion, "[%v]: %v", cluster.Meta.Name, logMessageKubernetesVersion)
-	}
+
+		createPostUpgradeWorkloads(u.T(), client, clusterName, cluster.FeaturesToTest)
```

Comment:
it would be nice to have an additional require and log that the k8s version is now the expected (updated) version

Reply:
Heard, will add it.

Reply:
Resolved